manifests/instances.pp

```diff
@@ -2,104 +2,86 @@
 #
 # [Remember: No empty lines between comments and class definition]
 class augeasproviders::instances (
-  $apache_directive_hash      = $augeasproviders::params::apache_directive_hash,
-  $apache_setenv_hash         = $augeasproviders::params::apache_setenv_hash,
-  $host_hash                  = $augeasproviders::params::host_hash,
-  $kernel_parameter_hash      = $augeasproviders::params::kernel_parameter_hash,
-  $mailalias_hash             = $augeasproviders::params::mailalias_hash,
-  $mounttab_hash              = $augeasproviders::params::mounttab_hash,
-  $nrpe_command_hash          = $augeasproviders::params::nrpe_command_hash,
-  $pam_hash                   = $augeasproviders::params::pam_hash,
-  $pg_hba_hash                = $augeasproviders::params::pg_hba_hash,
-  $puppet_auth_hash           = $augeasproviders::params::puppet_auth_hash,
-  $shellvar_hash              = $augeasproviders::params::shellvar_hash,
-  $ssh_config_hash            = $augeasproviders::params::ssh_config_hash,
-  $sshd_config_hash           = $augeasproviders::params::sshd_config_hash,
-  $sshd_config_subsystem_hash = $augeasproviders::params::sshd_config_subsystem_hash,
-  $sysctl_hash                = $augeasproviders::params::sysctl_hash,
-  $syslog_hash                = $augeasproviders::params::syslog_hash,
-  $resource_defaults          = $augeasproviders::params::resource_defaults
+  Hash $apache_directive_hash      = $augeasproviders::params::apache_directive_hash,
+  Hash $apache_setenv_hash         = $augeasproviders::params::apache_setenv_hash,
+  Hash $host_hash                  = $augeasproviders::params::host_hash,
+  Hash $kernel_parameter_hash      = $augeasproviders::params::kernel_parameter_hash,
+  Hash $mailalias_hash             = $augeasproviders::params::mailalias_hash,
+  Hash $mounttab_hash              = $augeasproviders::params::mounttab_hash,
+  Hash $nrpe_command_hash          = $augeasproviders::params::nrpe_command_hash,
+  Hash $pam_hash                   = $augeasproviders::params::pam_hash,
+  Hash $pg_hba_hash                = $augeasproviders::params::pg_hba_hash,
+  Hash $puppet_auth_hash           = $augeasproviders::params::puppet_auth_hash,
+  Hash $shellvar_hash              = $augeasproviders::params::shellvar_hash,
+  Hash $ssh_config_hash            = $augeasproviders::params::ssh_config_hash,
+  Hash $sshd_config_hash           = $augeasproviders::params::sshd_config_hash,
+  Hash $sshd_config_subsystem_hash = $augeasproviders::params::sshd_config_subsystem_hash,
+  Hash $sysctl_hash                = $augeasproviders::params::sysctl_hash,
+  Hash $syslog_hash                = $augeasproviders::params::syslog_hash,
+  Hash $resource_defaults          = $augeasproviders::params::resource_defaults
 ) inherits augeasproviders::params {
 
-  validate_hash($resource_defaults)
-
   if $apache_directive_hash and !empty($apache_directive_hash) {
```

Comment:
Since we're now defaulting to an empty hash, you can actually drop this line. `create_resources` can handle an empty hash just fine.

Reply:
Agreed and done